src/api/app/models/user.rb

```diff
@@ -840,6 +840,14 @@ def mark_login!
     update_attributes(last_logged_in_at: Time.now, login_failure_count: 0)
   end
 
+  def pending_announcement
+    Announcement.order(:created_at).last unless Announcement.last.in?(announcements)
```

Comment:
Why not `order(:id)` :thinking:

Reply:
There is no guarantee that the ids are assigned in a timewhise order

Reply:
Also why do you order the first statement but not the last? Shouldn't it be 
```
Announcement.order(:created_at).last unless Announcement.order(:created_at).last.in?(announcements)
```